doctr/__main__.py

```diff
@@ -237,15 +238,17 @@ def deploy(args, parser):
     build_repo = get_current_repo()
     deploy_repo = args.deploy_repo or build_repo
 
+    deploy_branch = 'master' if deploy_dir.endswith('github.io') else args.deploy_branch_name
```

Comment:
If someone passes the flag, that should be used regardless. But without the flag, the default should be `master` if the repo ends in `.github.io` or `.github.com` (you're missing the leading `.` by the way), and `gh-pages` otherwise.